src/main.c

```diff
@@ -3440,17 +3440,36 @@ int main(int argc, char **argv)
         conf.error_log = NULL;
     }
 
-    { /* raise RLIMIT_NOFILE */
-        struct rlimit limit;
+    {
+        struct rlimit limit = {0};
+
         if (getrlimit(RLIMIT_NOFILE, &limit) == 0) {
+            if (conf.max_connections > (int)limit.rlim_max) {
+                fprintf(stderr, "[error] The 'max-connections'=[%d] configuration value "
+                                "should not exceed the file descriptor limit of "
+                                "the process 'RLIMIT_NOFILE'=[%lu]\n",
+                        conf.max_connections, limit.rlim_max);
+                return EX_CONFIG;
+            }
+
+            /* raise RLIMIT_NOFILE */
             limit.rlim_cur = limit.rlim_max;
+
             if (setrlimit(RLIMIT_NOFILE, &limit) == 0
 #ifdef __APPLE__
                 || (limit.rlim_cur = OPEN_MAX, setrlimit(RLIMIT_NOFILE, &limit)) == 0
 #endif
-            ) {
+                ) {
                 fprintf(stderr, "[INFO] raised RLIMIT_NOFILE to %d\n", (int)limit.rlim_cur);
+            } else {
+                fprintf(stderr, "[warning] setrlimit(RLIMIT_NOFILE) failed "
+                                "with error %d:'%s'\n",
+                        errno, strerror(errno));
```

Comment:
`"[warning] setrlimit(RLIMIT_NOFILE) failed:%s", strerror(errno)` would be more consistent with other error messages, I think.